### PR TITLE
[AKS] AM-3basic filter resource with explicit access

### PR DESCRIPTION
**Pull request by zielinskikamil**
Added *filterResource* method to AM Service with tests & testing api

### CODE REVIEW

File: am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java [hunk 1]
```diff
@@ -17,4 +18,18 @@ public void createResourceAccess(String resourceId, String accessorId) {
...
+    /**
+     * Returns `resourceJSON` when row with userId (accessorId) and resourceId is in Db, otherwise null
+     * @param userId (accessorId)
+     * @param resourceId
+     * @param resourceJSON
+     * @return resourceJSON or null
+     */
+    public JsonNode filterResource(String userId, String resourceId, JsonNode resourceJSON) {
+        boolean hasAccess = jdbi.withExtension(AccessManagementRepository.class,
+                dao -> dao.explicateAccessExist(userId, resourceId));
```
Comment (arompa) [inline am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java]: is it explicate access on purpose or should it be explicit?

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/AccessManagementServiceIntegrationTest.java [hunk 2]
```diff
@@ -24,4 +31,23 @@ public void whenCreatingResourceAccess_ResourceAccessAppearsInDatabase() {
 
         assertThat(count).isEqualTo(1);
     }
+
+    @Test
```
Comment (arompa) [inline am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/AccessManagementServiceIntegrationTest.java]: 👍

File: am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java (hunk 1 above)
Comment (zielinskikamil) [reply am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java]: should be explicit

File: am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java [hunk 3]
```diff
@@ -17,4 +18,18 @@ public void createResourceAccess(String resourceId, String accessorId) {
         jdbi.useExtension(AccessManagementRepository.class,
             dao -> dao.createAccessManagementRecord(resourceId, accessorId));
     }
+
+    /**
+     * Returns `resourceJSON` when row with userId (accessorId) and resourceId is in Db, otherwise null
```
Comment (kpakur) [inline am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java]: The caller does not care where it's stored (db).
The message should be more meaningful for someone using this library what is this method for.

**Pull request merged**